os/board/rtl8730e/src/component/soc/amebad2/misc/ameba_tizenrt_pmu.c

```diff
@@ -240,12 +243,14 @@ void tizenrt_pre_sleep_processing(uint32_t *expected_idle_time)
 	/* ms =x*1000/32768 = (x *1000) >>15 */
 	ms_passed = (u32)((((u64)tick_passed) * 1000) >> 15);
 
+	u16 wakeup_reason = HAL_READ16(SYSTEM_CTRL_BASE_LP, REG_LSYS_DUMMY_090);
+
 #ifndef CONFIG_PLATFORM_TIZENRT_OS
 	vTaskStepTick(ms_passed); /*  update kernel tick */
 #else
 #ifdef CONFIG_PM_TICKSUPPRESS
 	if (tizenrt_sleep_handler) {
-		tizenrt_sleep_handler((u64)ms_passed); /*  update kernel tick */
+		tizenrt_sleep_handler((clock_t)ms_passed, (pm_wakeup_reason_code_t)wakeup_reason); /*  update kernel tick */
```

Comment:
void up_pm_board_sleep(void (*handler)(clock_t, pm_wakeup_reason_code_t))

Could you please change to use this handler?
we will remove up_register_wakehandler()
I added up_register_wakehandler temporary.

Reply:
done~